grails-app/services/com/unifina/service/ChallengeService.groovy

```diff
@@ -30,34 +23,18 @@ class ChallengeService {
 		}
 		boolean valid = address.toLowerCase() == publicKey.toLowerCase()
 		if (valid) {
-			getConnection().del(challengeID)
+			keyValueStoreService.delete(challengeID)
 		}
 		return valid
 	}
 
 	String verifyChallengeAndGetAddress(String challengeID, String challenge, String signature) {
-		String challengeRedis = getConnection().get(challengeID)
+		String challengeRedis = keyValueStoreService.get(challengeID)
 		boolean invalidChallenge = challengeRedis == null || challenge != challengeRedis
 		if (invalidChallenge) {
 			return null
-		} else {
-			getConnection().del(challengeID)
 		}
 		byte[] messageHash = ECRecover.calculateMessageHash(challenge)
```

Comment:
I was thinking of doing it in a separate branch after this one is merged because it is getting heavy. But if you prefer I can implement it in this branch.

Reply:
@mthambipillai It should be just a small change? Let's discuss when we meet today. I think we need to include it if that's what we're going for - otherwise we end up shipping this update and then making an incompatible change right after.